src/Compare.ts

```diff
@@ -0,0 +1,134 @@
+    }
+}
+
+export const handler = function (argv) {
+    let args = new IonCompareArgs(argv);
+    new Compare(args);
+}
+
+/** Comparison semantics to be used with compare command */
+export enum ComparisonType {
+    BASIC = "basic",
+    EQUIVS = "equivs",
+    NON_EQUIVS ="non_equivs",
+    EQUIVS_TIMELINE = "equivs_timeline"
+}
+
+/**
+ * ComparisonContext to create a structure for lhs and rhs streams
+ */
+export class ComparisonContext {
+    location: string;
+    eventStream: IonEventStream;
+
+    constructor(path: string, args: IonCompareArgs){
+        this.location = path;
+        let ionReader = this.createReadersForComparison(args);
+        this.setEventStream(ionReader, args);
+    }
+
+    createReadersForComparison(args: IonCompareArgs): Reader {
+        let ionReader;
+        try {
+            ionReader = makeReader(this.getInput(this.getLocation()));
+        } catch (Error) {
+            new IonCliError(ErrorType.READ, this.getLocation(), Error.message, args.getErrorReportFile()).writeErrorReport();
+        }
+        return ionReader;
+    }
+
+    getInput(path: string): string | Buffer {
+        let options = path.endsWith(".10n") ? null : "utf8";
+        return fs.readFileSync(path, options);
+    }
+
+    setEventStream(ionReader: Reader, args: IonCompareArgs) {
+        this.eventStream = new IonEventStream(ionReader, this.getLocation(), args);
+    }
+
+    getEventStream() {
+        return this.eventStream;
+    }
+
+    getLocation(): string {
+        return this.location;
+    }
+
+    writeComparisonContext(ionOutputWriter: Writer, isLHS: boolean, event_index: number) {
+        let field_name: string = isLHS ? "lhs" : "rhs";
+        ionOutputWriter.writeFieldName(field_name);
+        ionOutputWriter.stepIn(IonTypes.STRUCT);
+        ionOutputWriter.writeFieldName("location");
+        ionOutputWriter.writeString(this.location);
+        ionOutputWriter.writeFieldName("event");
+        this.getEventStream().getEvents()[event_index].write(ionOutputWriter);
+        ionOutputWriter.stepOut();
+        ionOutputWriter.writeFieldName("event_index");
+        ionOutputWriter.writeInt(event_index);
+    }
+}
+
+
+/**
+ * Compare all inputs (which may contain Ion streams and/or EventStreams) against all other inputs
+ * using the Ion data model's definition of equality. Write a ComparisonReport to the output.
+**/
+export class Compare {
+
+    constructor(parsedArgs: IonCompareArgs) {
+        let output_writer = OutputFormat.createIonWriter(parsedArgs.getOutputFormatName());
+        if (output_writer) {
+                this.compareFiles(output_writer, parsedArgs);
+        }
+    }
+
+    compareFiles(ionOutputWriter: Writer, args: IonCompareArgs): void {
+        for (let pathFirst of args.getInputFiles()) {
+            for(let pathSecond of args.getInputFiles()) {
+                if(pathFirst === pathSecond) {
```

Comment:
A path will still need to be compared against itself for equivs, non-equivs, and equiv-timeline semantics.